AM_CONDITIONAL must not be nested

--- origsrc/smuxi-0.8/configure.ac	2010-09-02 14:00:45.000000000 -0500
+++ src/smuxi-0.8/configure.ac	2011-01-09 23:52:35.846342300 -0600
@@ -144,7 +144,6 @@ PKG_CHECK_MODULES([LOG4NET], [log4net])
 PKG_CHECK_EXISTS([nini-1.1], FOUND_NINI=yes, FOUND_NINI=no)
 if test "x$FOUND_NINI" = "xyes"; then
 	nini_files=`pkg-config --variable=Libraries nini-1.1`
-	AM_CONDITIONAL([BUNDLE_NINI], test -n "$nini_files")
 	if test -z "$nini_files" ; then
 		# Debian-based distros place Nini into the GAC
 		PKG_CHECK_MODULES([NINI], [nini-1.1])
@@ -155,6 +154,7 @@ if test "x$FOUND_NINI" = "xyes"; then
 else
 	PKG_CHECK_MODULES([NINI], [nini >= 1.1])
 fi
+AM_CONDITIONAL([BUNDLE_NINI], test -n "$nini_files")
 
 # Optional Libraries
 AC_ARG_WITH([indicate],
